### PR TITLE
Move all store implementations into stores namespace

### PR DESCRIPTION
**Pull request by janjagusch**
I propose moving all stores into a common `stores` namespace,[ similarly to how `fsspec` does it](https://github.com/fsspec/filesystem_spec/tree/master/fsspec/implementations).

This change should be non-breaking, by keeping the old imports. Using the old import structure will trigger a deprecation warning, tough. Old imports can be deleted in `v2` release.

TODO:

- [ ] Change import structure in tests.
- [ ] Move some more modules into `stores`.

### CODE REVIEW

**Comment by janjagusch** [comment]
> Does this help in any other form?

@xhochy, this is a very opinionated PR, but I found the current structure very confusing when working on #45. I hope this refactoring will make developers more productive when working with the library. @simonbohnen, already seems to use the new import structure in #69.

i know this is a big refactoring. but given that all old imports still work (which should be tested), do you object to this change?

File: minimalkv/stores/_net_common.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# TODO: Rename this module.
```
Comment (janjagusch) [inline minimalkv/stores/_net_common.py:1]: This still needs to be done.

Comment (janjagusch) [reply minimalkv/stores/_net_common.py:1]: Given that the size of this PR and that this is a private module that we could rename at any time, I suggest we don't change anything about this module in this PR.

```suggestion
```

**Comment by xhochy** [comment]
> i know this is a big refactoring. but given that all old imports still work (which should be tested), do you object to this change?

Yes, I object. It is a big change and you can bikeshed a bit about the structure. I see benefits in the old one and in the new one, though neither of them is a clear winner. Large refactorings can introduce some issues later on (bugs ;)) and I don't fancy them in the chain (it has been quite fragile in the last weeks).

**Comment by simonbohnen** [comment]
> Yes, I object. It is a big change and you can bikeshed a bit about the structure. I see benefits in the old one and in the new one, though neither of them is a clear winner. Large refactorings can introduce some issues later on (bugs ;)) and I don't fancy them in the chain (it has been quite fragile in the last weeks).

Although I see the point that this is a big change, cleaning up the project structure would make future development easier (for me 😅). I think this might be inevitable if we intend to maintain and rely on this package long-term.

**Comment by xhochy** [comment]
I fail to see a good argument here. We have stores currently categorized by type and would now move to a single global directory for all of them. I see that it can be better in some cases but the same could be said for the current categorized approach.

**Comment by janjagusch** [comment]
> I fail to see a good argument here. We have stores currently categorized by type and would now move to a single global directory for all of them. I see that it can be better in some cases but the same could be said for the current categorized approach.

We could still group them by type, e.g. `from minimalkv.stores.memory import RedisStore`. What I think could be improved about the current package structure is that the `minimalkv` specification (and utilities) lives together with concrete store implementations inside the same namespace (e.g. `minimalkv.fs` and `minimalkv.decorator`).

**Comment by simonbohnen** [comment]
As this doesn't seem particularly pressing and we don't agree on a specific solution, should we close this for now?

**Comment by xhochy** [comment]
Yes, let's close it

**Pull request closed without merging**